### ISSUE TITLE
Alternative To Expensive Tri-Planar For Steep Slopes

### ISSUE THREAD

**Issue by LeeBamberTGC** [TheGameCreators/GameGuruRepo#2165]
One of many artifacts of the new terrain system is the stretching of terrain textures when subjected to extreme vertical height changes:

![image](https://user-images.githubusercontent.com/35840583/154485330-074e7f28-c202-4fdf-ac3e-d2f13cbd889e.png)

Modern games solve this in a number of ways, and we have used tri-planar in the past. TP can have a performance hit, and its nature requires the whole terrain shader is subject to the extra functionality.

An alternative idea is to use a batch of LOD-based static meshes that are superimposed onto the detected vertical drops, allowing the new geometry to have a higher density of vertices to avoid stretching and allow a different set of textures to be used specifically for this type of natural formation.

Almost vertical structures in nature are RARELY uniform, and an opportunity exists to create these mesh and texture forms either procedurally, or a selection created by an artist for different biome materials.  The above shot would benefit for example if the wall of ice looked like this:

![image](https://user-images.githubusercontent.com/35840583/154486003-6a81ff26-8732-4f4b-9d16-fdeaac27af11.png)

And grassy terrain would indicate strata and fallen debris at the base of such constructions:

![image](https://user-images.githubusercontent.com/35840583/154486174-11498e00-083e-4c8c-a751-dabef01576a9.png)

Thanks to instance batching, and LOD, with a texture plate set that can be re-used across different materials, the performance should be better than tri-planar, and offer a much higher quality rendering, not only from a distance but up-close too.  And all the while being generated automatically with no user intervention.

**Comment by ArgentArts** [comment]
Would such a system allow, in the future, for the end-user to create their own static-meshes to use for this for their own custom bioms? Say, for example, the end-user is creating an alien planet, etc.? This sounds great, to me, as long as, in the end, we are able to customize it.

**Comment by LeeBamberTGC** [comment]
@ArgentArts As we are at the discussion/planning stage, I think having customisability built in from the start is a nice idea, and in a way will help TGC make more biomes in the future a little bit easier (or more likely, sit back and watch the community make them for all the other biomes we have not dreamed up). Anyone fancy a MARS BIOME :)

**Comment by ArgentArts** [comment]
I agree. If it allows the end-user to make their own, then I am all for it. Else, it's just a nice feature, but one I won't use, to be frank. After all, I think that many of us want our games to be as unique as possible.

**Comment by UltraVox001** [comment]
I'm glad to read this post from @LeeBamberTGC .
Steep mountains and quality cliffs are nearly impossible to achieve with the current terrain editor. Unless you have to sacrifice quality, of course. Long live the procedural! I can not wait to see it. For the moment, I am content with classic (normal) grounds while I have many other ambitions...

@ArgentArts : I will use this function in either case. Not everyone knows how to make this type of asset, and even less when they are complex.

**Comment by ArgentArts** [comment]
Yes, this is useful. Of course. I just want to keep customization in the forefront. Why? Because when MAX started it was all about customization for adding your own trees, your own grass, your own ground textures, etc. And while that's going to happen, it all took a sideline for BIOMs ... BIOMs that are not, at the moment, customizable (or at least not easily customizable). And, yet, these bioms should be or could be easily customizable. So, I don't want to see yet another awesome feature added to MAX that is, yet again, awesome, but locked in such a way that is either not customizable or not easily customizable.
